lib/ansible/executor/action_write_locks.py

```diff
@@ -15,30 +15,44 @@
 # You should have received a copy of the GNU General Public License
 # along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
 
-# Make coding more python3-ish
-from __future__ import (absolute_import, division, print_function)
+from __future__ import absolute_import, division, print_function
+
 __metaclass__ = type
 
+import sys
 from multiprocessing import Lock
 
 from ansible.module_utils.facts.system.pkg_mgr import PKG_MGRS
+from ansible.utils.multiprocessing import context as multiprocessing_context
+
+try:
+    # Do not initialize this more than once.
+    action_write_locks
+except (NameError, UnboundLocalError):
+    action_manager = multiprocessing_context.Manager()
+    if sys.version_info > (3, 0, 0):
+        action_dict = action_manager.dict
+        action_lock = action_manager.Lock
+    else:
+        action_dict = dict
+        action_lock = Lock
 
-if 'action_write_locks' not in globals():
-    # Do not initialize this more than once because it seems to bash
-    # the existing one.  multiprocessing must be reloading the module
-    # when it forks?
-    action_write_locks = dict()
+    # These plugins are known to be called directly by action plugins
+    # with names differing from the action plugin name. We precreate
+    # them here as an optimization.
+    action_write_locks = action_dict(
+        copy=action_lock(),
+        file=action_lock(),
+        setup=action_lock(),
+        slurp=action_lock(),
+        stat=action_lock(),
+    )
 
     # Below is a Lock for use when we weren't expecting a named module.  It gets used when an action
     # plugin invokes a module whose name does not match with the action's name.  Slightly less
     # efficient as all processes with unexpected module names will wait on this lock
-    action_write_locks[None] = Lock()
+    action_write_locks[None] = action_lock()
 
-    # These plugins are known to be called directly by action plugins with names differing from the
-    # action plugin name.  We precreate them here as an optimization.
     # If a list of service managers is created in the future we can do the same for them.
-    mods = set(p['name'] for p in PKG_MGRS)
-
-    mods.update(('copy', 'file', 'setup', 'slurp', 'stat'))
```

Comment:
Why drop these?

Reply:
these were just added to the `action_dict` on it's initialization https://github.com/ansible/ansible/pull/74895/files#diff-b356a647a1cfc0932905abde09f8c03b2686464dd6036699a4b01aae7f62bc11R43 there's no need to do it more than once.